Bottom: eb1d840e1ad5d22219dc86ac7358eddcd76b7726
Top:    7bad16bc067b0682da90f1dadfb4355a246c3123
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-22 14:46:15 -0500

Fix mariadb bind ip address


---
diff --git a/roles/mariadb/tasks/main.yml b/roles/mariadb/tasks/main.yml
index 61b296f3..966c9a93 100644
--- a/roles/mariadb/tasks/main.yml
+++ b/roles/mariadb/tasks/main.yml
@@ -20,15 +20,14 @@
 - name: Open the firewall
   firewalld: service=mysql permanent=true state=enabled
   when: mariadb_open_firewall
-- name: Set host ssh ListenAddress
+- name: Set host mysqld bind-address
   template:
     src: mariadb-bind-address.j2
     dest: /etc/my.cnf.d/mariadb-bind-address.cnf
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
-
+  register: bind_cnf
 - name: Restart MariaDB service
   service:
     name: mariadb
     enabled: yes
     state: restarted
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
+  when: bind_cnf.changed
diff --git a/roles/mariadb/templates/mariadb-bind-address.j2 b/roles/mariadb/templates/mariadb-bind-address.j2
index f30ea73a..831edde0 100644
--- a/roles/mariadb/templates/mariadb-bind-address.j2
+++ b/roles/mariadb/templates/mariadb-bind-address.j2
@@ -1,3 +1,3 @@
 # Bind the database only to the host port
 [mysqld]
-bind-address = {{ ansible_br0["ipv4"]["address"] }}
+bind-address = {{ ansible_default_ipv4["address"] }}
